src/main/java/com/sandogh/sandogh/users/service/UserService.java

```diff
@@ -16,8 +24,29 @@
 public class UserService {
     private final UserDAO userDAO;
 
+    public List<User> getAllUser(){
+        return (List<User>) userDAO.findAll();
+    }
+
+    public void getDeleteUser(long id){
+        this.userDAO.deleteById(id);
+    };
+
+    public User getUserById(long id){
```

Comment:
set the method signature as the following:
`Optional<User> getUserById(long id)`

Do not handle not found exception here. DAO classes are not responsible for the business logic. Just use them for fetch/edit/insert/ or other DB actions.

Reply:
My suggested code would be:
`Optional<User> getUserById(long id){
   return userDAO.findById(id);
}`